base/doc/ltnews40.tex

```diff
@@ -151,6 +151,24 @@ \section{Code improvements}
 
 \section{Bug fixes}
 
+\subsection{Fix wrong file type in a rollback warning}
+
+When the \LaTeX{} is rolled back to date \meta{date1} and a class or
+package with minimum date requirement \meta{date2} is to be loaded, if
+\meta{date2} is later than \meta{date1}, a rollback warning is raised:
+\begin{verbatim}
+LaTeX Warning: Suspicious rollback/min-date date given
+               A minimal date of YYYY-MM-DD has been specified for package
+               '<pkgname>'.
+               But this is in conflict with a rollback request to YYYY-MM-DD.
+\end{verbatim}
+
+Before, wrong file type may be used in this message, for example
+\verb|document class '<pkgname>'| or \verb|package '<clsname>'|.
```

Comment:
In some cases this message showed a wrong file type or a wrong file name, i.e., ...

(actually I don't see how "package clsname could have been possible, so perhaps just

In some cases this message showed a wrong file type, i.e.,  "document class <pkgname>".

Reply:
> (actually I don't see how "package clsname could have been possible

```tex
\RequirePackage[2018/10/01]{latexrelease}
\documentclass{article}[2022/05/01]

\begin{document}
\end{document}
```
raises a warning containing "package 'article'".
```
LaTeX Warning: Suspicious rollback/min-date date given
               A minimal date of 2022/05/01 has been specified for package
               'article'.
               But this is in conflict with a rollback request to 2018/10/01.
```

That's because the old `\@cls@pkg` has only two branches and treats any non-`\@clsextension` `\@currext` as packages. In this example the `\@currext` is empty when the rollback warning is raised.

Reply:
thanks for setting me straight